### DISCUSSION TITLE
Internationalization of Messages in play 2.6

### DISCUSSION THREAD

**Discussion by nehapatkar14** [playframework/playframework#11468] [Q&A]
Hello , I am migrating from play 2.3 to 2.6 and getting many errors related to 
For example this is the code in my Application  controller
```java
public Result instruments(String filter) {
    try {
        return ok(instruments.render(InstrumentController.getInstrumentData(session("username"),
            session("primaryLabID"), getConnectionName()), filter, new ErrorInfo()));
    }
    catch (ControllerException e) {
        logger.error(e.getMessage(), e);
        return internalServerError(instruments.render(new InstrumentData(), filter, new ErrorInfo(e)));
    }
}
```
And my `instruments.scala.html` contains 
```
@(instrumentData: InstrumentData, filter: String, errorInfo: ErrorInfo, messages: Messages)
```
After changing my controller to following and keeping the instruments.scala as it is I am getting a new error . 
```java
@Security.Authenticated(Secured.class)
public Result instruments(String filter) {
    try {
        /**Messages message = Http.Context.current().messages();**/
        return ok(instruments.render(InstrumentController.getInstrumentData(session("username"),
            session("primaryLabID"), getConnectionName()), filter, new ErrorInfo(),/**message**/));
    }
    catch (ControllerException e) {
        logger.error(e.getMessage(), e);
        return internalServerError(instruments.render(new InstrumentData(), filter, new ErrorInfo(e)));
    }
}
```
```
Error:incompatible types: play.i18n.Messages cannot be converted to play.api.i18n.Messages
[ERROR]                     session("primaryLabID"), getConnectionName()), filter, new ErrorInfo(),message));
```
In controller I am using  the package : `play.i18n.Messages`
Which message parameter do I pass from my controller ?
Can someone help me with this migration ?

**Comment by tsuyoshizawa** [comment]
You need to use `MessageApi` from 2.6.x.

> The static deprecated methods in [play.i18n.Messages](https://www.playframework.com/documentation/2.8.x/api/java/play/i18n/Messages.html) have been removed in 2.6.x, as there are equivalent methods on the [MessagesApi](https://www.playframework.com/documentation/2.8.x/api/java/play/i18n/MessagesApi.html) instance.

https://www.playframework.com/documentation/2.8.x/MessagesMigration26

Here is a sample of how to use `MessageApi` in Java

https://github.com/playframework/play-samples/blob/2.8.x/play-java-fileupload-example/app/controllers/HomeController.java#L22

**Comment by jducoeur** [comment]
More generally, 2.3 to 2.6 is a *big* jump, reflecting many years of evolution of Play.  In the middle of that is the huge philosophical shift from globals and statics to doing everything with dependency injection.  I'd recommend doing that upgrade one major version at a time, and dealing with all the deprecations at each step before proceeding to the next.
